apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryOptions.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Google.Cloud.Storage.V1;
+
+/// <summary>
+/// 
+/// </summary>
+public sealed class RetryOptions
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public int MaxAttempts { get; set; }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public int InitialBackoff { get; set; }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public int MaxBackoff { get; set; }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public int BackoffMultiplier { get; set; }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public IList<int> ErrorCodes { get; set; }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public bool RetryAlways { get; set; }
```

Comment:
Rather than having `RetryAlways` and `ErrorCodes`, if you have a `Func<int, bool>` (or potentially a `Func<HttpResponseMessage, bool>` so that the predicate can check any aspect of the response) then you gain more flexibility.

You can then add factory methods/properties for `RetryAlways` (which would be `_ => true`) or a method to create a predicate from a set of error codes.